openlibrary/plugins/worksearch/code.py

```diff
@@ -1219,13 +1219,13 @@ def work_search(
             facet=facet,
             spellcheck_count=spellcheck_count,
         )
-        response = reply['response'] or ''
+        response = reply.get('response') or {}
     except (ValueError, OSError) as e:
         logger.error("Error in processing search API.")
         response = dict(start=0, numFound=0, docs=[], error=str(e))
 
     # backward compatibility
-    response['num_found'] = response['numFound']
+    response['num_found'] = response.get('numFound') or 0
```

Comment:
An alternative depending on how the caller reacts to 0 vs None.
```suggestion
    response['num_found'] = response.get('numFound') or None
```